### ISSUE TITLE
Поправить отображение устройств, проброшенных через бридж

### ISSUE THREAD

**Issue by rus-ik** [4mr/wb-engine#21]
Было бы здорово видеть имена устройств, проброшенных с другого контроллера через бридж, с именем коннекшена в префиксе..
Несколько вариантов как это можно реализовать.. напр где то указать путь к конфигу бриджа и оттуда взять имя из строки connection

Т.е. чтоб вместо:
![image](https://github.com/4mr/wb-engine/assets/743185/29aec96f-e8a5-4fac-8589-bb027115e527)

Стало: 
![image](https://github.com/4mr/wb-engine/assets/743185/681b3530-bf2a-433e-855e-822e11f6b6dd)

где WB1 - это имя коннекшена

**Comment by rus-ik** [comment]
Поясню почему это целесообразно..
HA на все хозяйство как правило один, а контроллеров бывает и больше чем один, поэтому было бы удобно собрать все топики там, где стоит wb-engine и отдать в HA..

**Comment by 4mr** [comment]
Не думаю, что получится реализовать такое. На каждом контроллере может быть свой wb-engine, также HA можно несколько поставить, они объединяются удобно.

**Comment by rus-ik** [comment]
Два HA это лишнее использование ресурсов и без того не очень производительных контроллеров..
проще, по-моему,  просто по tcp передавать топики в одну точку, чем поднимать два тяжелых HA..

**Comment by 4mr** [comment]
Скрипт wb-engine задумывался как посредник между MQTT контроллера и HA используя механизм autodiscovery, что либо получать по tcp там нет возможности, возможности wb-rules ограничены. В целом код открыт, можете попробовать реализовать и поделиться с сообществом.

**Comment by rus-ik** [comment]
Нет, подождите, видимо  я не совсем ясно объяснил.. топики по tcp собираются в один контроллер брокерами mqtt, штатными средствами, никакие продукты wb тут не нужны..

текущий wb-engine их прекрасно отображает, проблема только в том что имена топиков в wb-engine перезаписываются..

смотрите..есть контроллеры wb1 и wb-t14-1, бриджем топики /devices/wb-gpio c wb-t14-1 приходят на wb1..
как пример, скажем берем 14ый вх/выход модуля WBIO-DI-WD-14, который подключени и к wb1 и к wb-t14-1
на wb1 их теперь два:

вот один первый, с локально подключенного wd14
![image](https://github.com/4mr/wb-engine/assets/743185/9ea7e246-4f44-4fb8-a4e1-4716462f39ca)

вот второй, с удаленного, подключенного к wb-t14-1
![image](https://github.com/4mr/wb-engine/assets/743185/7ee9db2b-28fa-4a12-ba64-e7ec167fc804)

но в wb-engine отображается только один EXT1_IN14:
![image](https://github.com/4mr/wb-engine/assets/743185/be907a98-a0b7-4ed7-a413-8036f79988f9)

поэтому я и предложил добалять префикс..
по сути там нужно просто учесть, что топики могут приходить через бридж, вот и все..
и если как опцию добавить путь к конфигу бриджа, то получится все отобразить с корректными префиксами

**Comment by 4mr** [comment]
Покажите как выглядит конфиг бриджинга? 
Дело в том, что по имени устройства из топика определяется в том числе тип устройства. (/devices/<имя>/controls/EXT1_IN14)

**Comment by rus-ik** [comment]
В вышеописанном случае вот так:

$cat /etc/mosquitto/conf.d/20bridges.conf 
connection wb-t34-1
address 192.168.xx.xx
notifications true
notification_topic /wb-t34-1/bridge_status
keepalive_interval 20
restart_timeout 20

topic /# in 1 /devices/wb-t34-1 /devices/wb-gpio

**Comment by rus-ik** [comment]
Это конфиг на стороне wb1

**Comment by 4mr** [comment]
> topic /# in 1 /devices/wb-t34-1 /devices/wb-gpio

Получается так вы дубли только wb-gpio убрали, но там же много еще устройств с неуникальным названием и они будут дублироваться и значения wb1 будут перетираться с wb2.

**Comment by rus-ik** [comment]
Да, я знаю.
Собственно именно поэтому я и предложил добавлять имя коннекта, в случае работы с бриджем

**Comment by 8bitjoey** [comment]
Не читал весь тред, но я тоже столкнулся с необходимостью работать с несколькими WB и одним HA.
Пришлось внести некоторые правки в код, чтобы все заработало как мне хотелось:
https://github.com/4mr/wb-engine/pull/29

Скрипт работает точно так же, как и раньше, но позволяет задать префиксы для mqtt топиков, по которым HA будет управлять устройством (через wb_topic_prefix). А также позволяет задать некий префикс для device_id и unique_id, чтобы избежать возможных коллизий между разными WB.

Проверено локально, работает отлично.
